.github/workflows/python.yaml

```diff
@@ -0,0 +1,221 @@
+name: Conda Package and Python Formatting
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+jobs:
+  cancel_previous:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: styfle/cancel-workflow-action@0.6.0
+      if: ${{ github.event_name == 'pull_request' }}
+      with:
+        access_token: ${{ github.token }}
+
+  build_and_package:
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os:
+        - ubuntu-18.04
+        # - macOS-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
+        fetch-depth: 0
+
+    ###
+    # Caching between builds
+    ###
+    - name: Cache conda packages
+      uses: actions/cache@v2
+      with:
+        key: ${{matrix.os}}-conda-pkgs-${{hashFiles('**/conda_recipe/meta.yaml')}}
+        path: |
+            ~/miniconda/pkgs/*.tar.bz2
+            !~/miniconda/pkgs/galois-*
+    - name: Cache miniconda
+      id: cache-miniconda
+      uses: actions/cache@v2
+      with:
+        key: ${{matrix.os}}-miniconda-${{hashFiles('.github/**')}}
+        path: ~/.cache/miniconda
+    - name: Cache ccache objects
+      uses: actions/cache@v2
+      with:
+        key: ${{matrix.os}}-ccache-${{github.sha}}
+        restore-keys: |
+          ${{matrix.os}}-ccache-
+        path: ~/.ccache
+    - name: Setup ccache
+      run: |
+        ccache --max-files 600
+    - name: Cache graph inputs
+      id: cache-graph
+      uses: actions/cache@v2
+      with:
+        key: generic-cache-graph-${{hashFiles('.github/**')}}
+        path: ~/.cache/graph
+
+    ###
+    # Install platform-level dependencies (OS-specific)
+    ###
+    - name: Install dependencies
+      if: startsWith(matrix.os, 'ubuntu-')
+      run: |
+        sudo apt install -yq ccache curl libz-dev bzip2 make libnuma1
+    - name: Install dependencies
+      if: startsWith(matrix.os, 'macOS-')
+      run: |
+        brew install ccache curl
+      # TODO: Does brew install need bzip2, libz-dev, and make?
+    
+    ###
+    # Install Conda environment
+    ###
+    - name: Download miniconda
+      run: |
+        MINICONDA_FILE="$HOME/.cache/miniconda/miniconda.sh"
+        case ${{matrix.os}} in
+          ubuntu-*) PLAT=Linux-x86_64 ;;
+          macOS-*)  PLAT=MacOSX-x86_64 ;;
+          *)        echo "Unknown OS"; exit 10 ;;
+        esac
+
+        INPUT_URL="https://repo.anaconda.com/miniconda/Miniconda3-py38_4.8.3-${PLAT}.sh"
+        mkdir -p "$(dirname $MINICONDA_FILE)"
+        curl -fL --output "$MINICONDA_FILE" "$INPUT_URL"
+      if: steps.cache-miniconda.outputs.cache-hit != 'true'
+    - name: Download graphs
+      run: |
+        GRAPH_FILE="$HOME/.cache/graph/inputs.tar.gz"
+        GRAPH_LOCATION="$HOME/.cache/graph/inputs"
+        INPUT_URL="https://katana-ci-public.s3.us-east-1.amazonaws.com/inputs/katana-inputs-v14.tar.gz"
```

Comment:
IIRC, there is some code in python codebase that can download inputs on demand. If that's true, I might consider just always relying on that rather using all this cache logic because downloading the inputs takes a relatively small amount of time anyway.